raiden-dapp/src/store/index.ts

```diff
@@ -156,13 +160,16 @@ const store: StoreOptions<RootState> = {
       );
     },
     allTokens: (state: RootState): Token[] =>
-      Object.values(state.tokens).sort((a: Token, b: Token) => {
-        if (hasNonZeroBalance(a, b)) {
-          return (b.balance! as BigNumber).gt(a.balance! as BigNumber) ? 1 : -1;
-        }
-
-        return a.symbol && b.symbol ? a.symbol.localeCompare(b.symbol) : 0;
-      }),
+      Object.values(state.tokens)
+        .sort((a: Token, b: Token) => {
+          if (hasNonZeroBalance(a, b)) {
+            return (b.balance! as BigNumber).gt(a.balance! as BigNumber)
+              ? 1
+              : -1;
+          }
+          return a.symbol && b.symbol ? a.symbol.localeCompare(b.symbol) : 0;
+        })
+        .filter(token => state.tokenAddresses.includes(token.address)),
```

Comment:
Wouldn't it be better to first filter and then sort? 

In this case, it is just an extra item so it is not too big of an issue, but in general, I would put the filter first to avoid sorting items that won't be needed

Reply:
Alrighty! Thx for that feedback.